### ISSUE TITLE
[508-defect-1]: Dismissible alert on the benefit letter download page doesn't work and isn't uniquely labelled

### ISSUE THREAD

**Issue by joshkimux** [department-of-veterans-affairs/va.gov-team#45641]
## What happened?
The dismissible alert "Download your VA Letters on your mobile device" does not respond to keyboard (space and enter) and mouse (click). The button's label is also generic "Close notification." 

See @bkjohnson 's comment below!

## Specs:
- Device: Macbook
- Browser: Chrome and safari

## Steps to Reproduce
- URL: https://staging.va.gov/records/download-va-letters/

## Desired behavior
- This should respond to `space`, `enter`, and `click`
- This can be uniquely labelled by either using:
  - Updating the `aria-label` e.g. aria-label="Close download VA letters notification"
  - Using `aria-describedby` to let the heading describe the button

## Acceptance Criteria
- [ ] Button responds to `space`, `enter`, and `click`
- [ ] Button label is unique 

## How to configure this issue
- [x] **Attached to a Milestone** (when will this be completed?)
- [x] **Attached to Epic** (what body of work is this a part of? possibly `Ongoing Maintenance`)
- [x] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `triage`, `tools-improvements`)
- [x] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [x] **Labeled with `Bug`**

**Comment by bkjohnson** [comment]
@joshkimux the change I was calling out is `onClose` to `onCloseEvent` - the contents of the handler can probably remain the same. We have [documentation on events that fire](https://design.va.gov/storybook/?path=/docs/components-va-alert--default) from the component so that they can be handled:

![A screenshot of the storybook page for the alert component. The `closeEvent` is highlighted, reading "Fires when the component is closed by clicking on the close icon. This fires only when closeable is true."](https://user-images.githubusercontent.com/2008881/184400342-93ca27bd-9d74-4f16-81ba-d68322b4a9ea.png)

**Comment by joshkimux** [comment]
Thank you for clarifying! I've updated the ticket to just point to this comment 😆 🙏

**Comment by bkjohnson** [comment]
Some more `:two-cents:` - the way [this code is hiding the alert component by not rendering it](https://github.com/department-of-veterans-affairs/vets-website/blob/57658f5846ee378309cc774133ef7e1aabd41db2/src/applications/static-pages/letters-mobile-cta/components/App/index.js#L49-L57) might not be great for a11y reasons, so I'm eager to get your input Josh. As you can see from the link above, we provide the `visible` prop, and that's been the established way of dismissing something. When `visible` is falsey, [the contents of the alert is replaced with ` <div aria-live="polite" />` ](https://github.com/department-of-veterans-affairs/component-library/blob/3fc3b7d33d9cfe5ac51a54d7aed0b445c26cfe05/packages/web-components/src/components/va-alert/va-alert.tsx#L162).

You can see something similar here [with the Modal React component](https://github.com/department-of-veterans-affairs/vets-website/blob/5f4626f86c9ed5db57b89ee09f9f4b3d5d867ba5/src/applications/edu-benefits/components/OptOutWizard.jsx#L10-L33). Closing the modal will change a state value which is passed to the `visible` prop.

If that pattern is being used incorrectly for the alert component please let us know.

**Comment by joshkimux** [comment]
@bkjohnson so glad you pointed this out! It definitely explains alot of the pain points we've had in the past where we've had to set up timeouts for `aria-live` to work 😓 

Ideally, we'd want to follow `aria-live` documentation.

For alerts that dynamically appear on the page after being prompted by a button or selection, we'd probably want those alert's containers to exist ahead of time on page load with `aria-live="polite"` _and then_ have any dynamically "appearing" content be added into the empty container.

Static alerts like this one that exist on page load probably don't require the `aria-live` attribute at all as nothing is actually being updated on the page.

I'd love to work with @SarahKay8 and @davidakennedy to see if we can start looking into systemic fixes for this 🙏

**Comment by laflannery** [comment]
Regarding ownership and which team should handle this, I tracked this down a bit and this was originally a Public Websites (an older team but it was PW) issue. We had originally added this react widget but now the file is officially owned by Benefits Team, the product documentation is here:

- [VFS Product Directory: Letters](https://depo-platform-documentation.scrollhelp.site/getting-started/letters)
- [GitHub Documentation: Benefit Letters](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/products/benefit-letters)

There was also a question about if this particular alert should be dismissible or not - and that answer would direct how to resolve this. When the old Public Websites team added this, they made it dismissible because they were copying the implementation of the alert on the Claim Status Tool page. [This is the ticket for the original request.](https://github.com/department-of-veterans-affairs/va.gov-team/issues/35382). However, the alert seems to have been removed from the Claim Status Tool page (@joshkimux might be looking into this because perhaps that was not intentional?) but that means that perhaps we don't need to match behavior anymore and can choose to make this alert not dismissible.

Let me know if there are further questions on the background/ownership that I could answer, thanks!
